snapshot-labs/sx-monorepo
# feat: store per-treasury execution on proposals

### Summary

oSnap proposals can have separate executions for multiple treasuries.
This commit will display those in the UI as separate intities, linked to specific treasury.

Currently we read treasuries from latest metadata, this means that if one treasury is used in proposal, but then it's removed or updated when you go back to the proposal it will either show Unnamed treasury or updated name.
We should store at-proposal-time treasury details on proposal directly so we have correct version, but for now I skipped this because of TheGraph's issue we can't deploy new subgraphs.

### How to test

0. Run `VITE_ENABLED_NETWORKS=s-tn,sep yarn dev`.
1. Check two proposals (onchain and off-chain).
2. **onchain** http://localhost:8080/#/sep:0xfa463A2Bdd8d27f1b0eA99270c3c4Bb3A7B2ADeF/proposal/1
3. **onchain** http://localhost:8080/#/s-tn:0cf5e.eth/proposal/0x18b4c85fb3dba579b2714a3fcdba5ea96088f9f0a51f02002e81cc39a589d674
4. They look correct.

### Screenshots

<img width="726" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/1968722/5fe2c2ea-8246-4ccb-b749-1ec4dde7cd8c">
<img width="662" alt="image" src="https://github.com/snapshot-labs/sx-monorepo/assets/1968722/82c7f4ee-6db0-49ee-ae9c-1e693bfdf6bf">
